### PR TITLE
Feature/1108 - Asset Folder Creator MCP process

### PR DESCRIPTION
**Pull request by davidjgonzalez**
Similar to Tag Creator, but for creating Asset Folder structures with friendly (jcr:title) titles.

### CODE REVIEW

**Comment by davidjgonzalez** [comment]
Will fix code climate

File: bundle/src/test/java/com/adobe/acs/commons/mcp/impl/processes/AssetFolderCreatorTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
+package com.adobe.acs.commons.mcp.impl.processes;
```
Comment (justinedelson) [inline bundle/src/test/java/com/adobe/acs/commons/mcp/impl/processes/AssetFolderCreatorTest.java]: missing header

File: bundle/src/test/java/com/adobe/acs/commons/mcp/impl/processes/AssetFolderDefinitionTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package com.adobe.acs.commons.mcp.impl.processes;
```
Comment (justinedelson) [inline bundle/src/test/java/com/adobe/acs/commons/mcp/impl/processes/AssetFolderDefinitionTest.java]: missing header

File: bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/AssetFolderCreator.java [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+                folder = resourceResolver.create(resourceResolver.getResource(assetFolderDefinition.getParentPath()),
+                        assetFolderDefinition.getName(),
+                        folderProperties);
+
+                status = ReportRowStatus.CREATED;
+            } else {
+                status = ReportRowStatus.UPDATED_FOLDER_TITLES;
+            }
+
+            final Resource jcrContent = folder.getChild(JcrConstants.JCR_CONTENT);
```
Comment (badvision) [inline bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/AssetFolderCreator.java]: This kind of null checking makes me pine for a JCR API that embraces the new Optional stuff.  But I digress, this looks fine.

**Comment by ahmed-musallam** [comment]
@davidjgonzalez you sir, are a genius!
This does not override folders if they exist, only updates the title!! Thank you for saving my life!

**Comment by badvision** [comment]
He is a genius indeed!!

**Pull request merged**